src/AzureDeprecation.APIs.REST/Utils/DeprecationRequestModelBinder.cs

```diff
@@ -0,0 +1,126 @@
+using System.ComponentModel.DataAnnotations;
+using System.Reflection;
+using AzureDeprecation.APIs.REST.DataAccess.Models;
+using CodeJam;
+using CodeJam.Collections;
+using CodeJam.Reflection;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+
+namespace AzureDeprecation.APIs.REST.Utils;
+
+public static class DeprecationRequestModelBinder
```

Comment:
Can you elaborate on where we use this and for what purpose lease?

Reply:
Oh I have just seen `var filter = DeprecationRequestModelBinder.CreateModel(request.Query);`

Why can't we just use this for every property?

`var something = request.Query.TryGetValue("name", out var n) ? n.FirstOrDefault() : null;`

Reply:
To have more universal code. To add new filter, with current approach, you can just add property to filter model.
BTW, don't have a lot of reasons to do it this way

Reply:
Isn't this a bit overengineered? Also, if we decide to rename a property it would cause a breaking change automatically (without us actually knowing)?

Reply:
Let's refactor later on